src/main/java/Jarvis/Task.java

```diff
@@ -0,0 +1,52 @@
+package Jarvis;
+
+/**
+ * Abstract class representing a task.
+ * <p>
+ *     This class serves as the parent for different types of tasks to inherit from and encapsulates
+ *     the basic information and operations related to a task.
+ * </p>
+ */
+public class Task {
```

Comment:
Do consider adding the abstract keyword to abstract classes and methods so it is more obvious to others as well!